README.md

```diff
@@ -15,251 +18,329 @@ The role can be used to configure:
+* `wait: <seconds>` is a timeout that enables you to decide how long you give the device to
+activate. The default is using a suitable timeout. Note that the `wait` option is
+only supported by NetworkManager.
 
-```yaml
-network_connections:
-  - name: eth0
-    persistent_state: absent
-```
+Note that `state: up` always re-activates the profile and possibly changes the
+networking configuration, even if the profile was already active before. As
+a consequence, `state: up` always changes the system.
 
-Above example ensures the absence of a connection profile. If a profile with `name` `eth0`
-exists, it will be deleted.
+#### `state: down`
 
-* For NetworkManager this deletes all connection profiles with the matching `connection.id`.
-  Deleting a profile usually does not change the current networking configuration, unless
-  the profile was currently activated on a device. In that case deleting the currently
-  active connection profile disconnects the device. That makes the device eligible
-  to autoconnect another connection (see also [rh#1401515](https://bugzilla.redhat.com/show_bug.cgi?id=1401515)).
+- For `NetworkManager`,  it corresponds to `nmcli connection id {{name}} down`.
 
-* For initscripts it results in the deletion of the ifcfg file. Usually that
-  has no side-effect, unless some component is watching the sysconfig directory.
+- For `initscripts`, it corresponds to call `ifdown {{name}}`.
 
-#### Example
+You can deactivate a connection profile, even if is currently not active. As a consequence, `state: down` always changes the system.
 
-```yaml
-network_connections:
-  - name: eth0
-    #persistent_state: present  # default
-    type: ethernet
-    autoconnect: yes
-    mac: 00:00:5e:00:53:5d
-    ip:
-      dhcp4: yes
-```
+Note that if the `state` option is unset, the connection profile’s runtime state will not be changed.
 
-Above example creates a new connection profile or ensures that it is present
-with the given configuration. It implies the `persistent_state` setting to be
-`present`.
 
-Valid values for `type` are:
+### `persistent_state`
+
+The `persistent_state` option identifies if a connection profile is in a persistent state. The `persistent_state` option can be set to the following values:
+
+#### `persistent_state: present` (default)
+
+Note that if `persistent_state` is `present` and the connection profile contains
+the `type` option, the profile will be created or updated. If the connection profile is
+incomplete (no `type` option), the behavior is undefined. Also, the `present` value
+does not directly result in a change in the network configuration. If the `state` option
+is not set to `up`, the profile is only created or modified, not activated.
+
+#### `persistent_state: absent`
+
+The `absent` value ensures that the profile is not present on the target host. For
+example, if a profile with `name` `eth0` exists, it will be deleted. In this case:
```

Comment:
The matching example configuration was moved to a different place, so this part become unintelligible (it is not clear why one mentions `eth0` here).